x/programs/rust/examples/token/src/lib.rs

```diff
@@ -1,22 +1,17 @@
-use wasmlanche_sdk::program::Program;
 use wasmlanche_sdk::store::Context;
-use wasmlanche_sdk::types::{Address, Bytes32};
+use wasmlanche_sdk::types::Address;
 
 use expose_macro::expose;
 
 /// Initializes the program with a name, symbol, and total supply.
 #[expose]
-pub fn init_program() -> i64 {
-    let mut token_program = Program::new();
-    token_program
-        .add_field(String::from("name"), Bytes32::from("WasmCoin".to_string()))
+pub fn init(ctx: Context) -> bool {
+    ctx.store_value("total_supply", &123456789_i64)
```

Comment:
For another PR, but could use some type of builder pattern here to clean this up such that the code looks like this:
```rust
ctx
    .store_value("total_supply", &123456789_i64)
    .store_value("name", "WasmCoin")
    .store_value("symbol", "WACK")
    .is_ok()
```
where 
```rust
impl Context {
    // not sure what the actual signature is, but I think you can get the gist from my example
    fn store_value<K:Serialize, V: Serialize>(&self, key: K, value: V) -> StoreResult<'_> {
        match store_value(key, value) {
            Ok(_) => StoreResult::Ok(&ctx),
            Err(e) => StoreResult::Err(e),
        }
    }
}

enum StoreResult<'a> {
    Ok(&'a ctx),
    Err(Error),
}

impl StoreResult<'_> {
    fn store_value(self, key: K, value: V) -> Self {
        match self {
            Self::Ok(ctx) => ctx.store_value(k, v),
            err => err,
        }
    }
}

Reply:
added this to this pr, makes it much cleaner. however we have to implement the Result<> functions like `is_ok` and `is_err`.

Reply:
Can just have a `into_result` method instead. But save that for another PR if you want to do it. Feel free to resolve this comment.